bioimageio/core/contrib/contrib_a/_inference.py

```diff
@@ -0,0 +1,199 @@
+import collections
+from os import PathLike
+from typing import Dict, IO, List, Optional, OrderedDict, Sequence, Tuple, Union
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+
+from bioimageio.core.contrib.utils import (
+    get_chunk,
+    get_corrected_chunks,
+    get_default_input_tile,
+    get_output_rois,
+    transpose_sequence,
+    tuple_roi_to_slices,
+)
+from bioimageio.core.prediction_pipeline._combined_processing import CombinedProcessing
+from bioimageio.core.prediction_pipeline._model_adapters import ModelAdapter, create_model_adapter
+from bioimageio.core.resource_io import nodes
+from bioimageio.core.resource_io.utils import resolve_raw_node
+from bioimageio.spec import load_raw_resource_description
+from bioimageio.spec.model import raw_nodes
+from bioimageio.spec.shared.raw_nodes import ResourceDescription as RawResourceDescription
+
+try:
+    from typing import Literal
+except ImportError:
+    from typing_extensions import Literal
+
+
+BoundaryMode = Literal["reflect"]
+
+
+def forward(*tensors, model_adapter: ModelAdapter, output_tile_roi: Tuple[slice, ...]):
+    """helper to cast dask array chunks to xr.DataArray and apply a roi to the output"""
+    assert len(model_adapter.bioimageio_model.inputs) == len(tensors), (
+        len(model_adapter.bioimageio_model.inputs),
+        len(tensors),
+    )
+    tensors = [xr.DataArray(t, dims=tuple(ipt.axes)) for ipt, t, in zip(model_adapter.bioimageio_model.inputs, tensors)]
+    output = model_adapter.forward(*tensors)[0]  # todo: allow more than 1 output
+    return output[output_tile_roi]
+
+
+async def run_model_inference_with_dask(
+    model_rdf: Union[str, PathLike, dict, IO, bytes, raw_nodes.URI, RawResourceDescription],
```

Comment:
@oeway for this workflow decoding lazy xarray arrays (dask arrays) would be great!